odoo/fields.py

```diff
@@ -1331,19 +1332,31 @@ def recompute(self, records):
         if not to_compute_ids:
             return
 
+        def missing_guard(func, batch_records):
```

Comment:
Maybe this one could be a field method, like `field.compute_except_missing(records)`, which calls `field.compute` internally.  Notice that `missing_guard` can only be called with `field.compute_value` or something equivalent to that.